File: 03_Sorting Techniques/02_BubbleSort/02_BubbleSort.cpp
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;

void bubbleSort(vector<int> &nums) {
    for (int i = 0; i < nums.size() - 1; i++) {
        int max = nums[i];
        for (int  j = 0; j < nums.size() - 1 - i; j++ ) {
            if (nums[j] > nums[j+1]) swap(nums[j], nums[j+1]);
        }
    }
}


int main() {
    vector<int> arg = {-1, 100, 12, 123, 32, -32, 100};
    bubbleSort(arg);
    for (auto x: arg) {
        cout << x << ' ';
    }
    return 0;
}

File: 03_Sorting Techniques/02_BubbleSort/02_BubbleSort.js
function bubbleSort(nums) {
    for (let i = 0; i < nums.length - 1; i++) {
        for (let j = 0; j < nums.length - i - 1; j++) {
            if (nums[j] > nums[j+1]) {
                [nums[j], nums[j+1]] = [nums[j+1], nums[j]];
            }
        }
    }
    return nums;
}

console.log(bubbleSort([-1,22,1,100,23,-11,0]));